examples/digits_dann_lightn/main.ipynb

```diff
@@ -0,0 +1,191 @@
+   "id": "8253f704",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "cfg_path = \"./configs/QUICK-AND-DIRTY.yaml\"\n",
+    "gpus = None"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b05cf0fb",
+   "metadata": {},
+   "source": [
+    "# ---- setup configs ----"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "ded6b2d1",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "cfg = get_cfg_defaults()\n",
+    "cfg.merge_from_file(cfg_path)\n",
+    "cfg.freeze()\n",
+    "print(cfg)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "759d55dc",
+   "metadata": {},
+   "source": [
+    "# ---- setup output ----"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "292a0aac",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "os.makedirs(cfg.OUTPUT.DIR, exist_ok=True)\n",
+    "format_str = \"@%(asctime)s %(name)s [%(levelname)s] - (%(message)s)\"\n",
+    "logging.basicConfig(format=format_str)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2a9d0bcd",
+   "metadata": {},
+   "source": [
+    "# ---- setup dataset ----"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "df1ae7ac",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "source, target, num_channels = DigitDataset.get_source_target(\n",
+    "    DigitDataset(cfg.DATASET.SOURCE.upper()), DigitDataset(cfg.DATASET.TARGET.upper()), cfg.DATASET.ROOT\n",
+    ")\n",
+    "dataset = MultiDomainDatasets(\n",
+    "    source,\n",
+    "    target,\n",
+    "    config_weight_type=cfg.DATASET.WEIGHT_TYPE,\n",
+    "    config_size_type=cfg.DATASET.SIZE_TYPE,\n",
+    "    val_split_ratio=cfg.DATASET.VAL_SPLIT_RATIO,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "71b96eb0",
+   "metadata": {},
+   "source": [
+    "# Repeat multiple times to get std"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "66cc2485",
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "for i in range(0, cfg.DATASET.NUM_REPEAT):\n",
+    "    seed = cfg.SOLVER.SEED + i * 10\n",
+    "    # seed_everything in pytorch_lightning did not set torch.backends.cudnn\n",
+    "    set_seed(seed)\n",
+    "    print(f\"==> Building model for seed {seed} ......\")\n",
+    "    # ---- setup model and logger ----\n",
+    "    model, train_params = get_model(cfg, dataset, num_channels)\n",
+    "    logger, results, checkpoint_callback, test_csv_file = setup_logger(\n",
+    "        train_params, cfg.OUTPUT.DIR, cfg.DAN.METHOD, seed\n",
+    "    )\n",
+    "\n",
+    "    if gpus is None:\n",
+    "        trainer = pl.Trainer(\n",
+    "            progress_bar_refresh_rate=cfg.OUTPUT.PB_FRESH,  # in steps\n",
+    "            min_epochs=cfg.SOLVER.MIN_EPOCHS,\n",
+    "            max_epochs=cfg.SOLVER.MAX_EPOCHS,\n",
+    "            checkpoint_callback=checkpoint_callback,\n",
```

Comment:
Is your Pytorch Lightning>=1.3.0? Could you try replacing `checkpoint_callback=checkpoint_callback` with `callbacks=[checkpoint_callback],`. It probably can fix the above Attribute Error. Reference: https://pytorch-lightning.readthedocs.io/en/stable/common/weights_loading.html

Reply:
Thanks @sz144 - I'll check this out.

Reply:
This has helped and the notebook now runs, with Pytorch Lightning>=1.3.0 specified in `setup.py` for extras.

Reply:
@haipinglu we need check our previous examples for compatibility with the latest PyTorch Lightning.

Reply:
Or we could pin a specific earlier version.

Reply:
Ideally, this should be covered by `tests` rather than manual check. Of course examples are not covered by tests but I think good tests should do this. The priority should be the API, then if possible we could cover the examples. But I do not think that this is the priority now.

If we found any compatibility issues, we can have respective tests to cover them or discuss how to prevent that from happening in future.